src/mlpack/methods/ann/init_rules/variance_scaling_init.hpp

```diff
@@ -0,0 +1,202 @@
+    init.Initialize(W, rows, cols);
+  }
+
+  template<typename eT, typename Mode = ModeType>
+  void Initialize(
+      arma::Mat<eT>& W,
+      const size_t rows,
+      const size_t cols,
+      typename std::enable_if_t<std::is_same<Mode, FanAvg>::value>* = 0)
+  {
+    double var = (2 * scalingFactor / ((double) (rows) + (double) (cols)));
+    GaussianInitialization init(0, var);
+    init.Initialize(W, rows, cols);
+  }
+
+
+  /**
+   * Initialize the elements of the specified weight 3rd order tensor.
+   *
+   * @param W Weight matrix to initialize.
+   * @param rows Number of rows.
+   * @param cols Number of columns.
+   * @param slice Numbers of slices.
+   */
+  template<typename eT>
+  void Initialize(arma::Cube<eT>& W,
+                  const size_t rows,
+                  const size_t cols,
+                  const size_t slices)
+  {
+    W = arma::Cube<eT>(rows, cols, slices);
+
+    for (size_t i = 0; i < slices; i++)
+      Initialize(W.slice(i), rows, cols);
+  }
+
+ private:
+  const size_t scalingFactor;
+}; // class VarianceScalingNormalInit
+
+
+/**
+ * This class is used to initialize weight matrix from a random distribution
+ * with bound = [-limit,limit] where limit is decided by mode type. The limit
+ * is given by sqrt(3*scaling factor/n) where n is as follows :
+ * n = rows for FanIn mode (number of neurons feeding in)
+ * n = cols for FanOut mode (number of neurons feeding out)
+ * n = (rows + cols)/2 for FanAvg mode (average of neurons feeding in and out)
+ */
+template<typename ModeType = FanAvg>
+class VarianceScalingUniformInit
+{
+ public:
+  // Empty Constructor
+  // Scaling factor is the amount by which bound will be scaled.
+  VarianceScalingUniformInit(const size_t scalingFactor = 1):
+  scalingFactor(scalingFactor)
+  {}
+
+  template<typename eT, typename Mode = ModeType>
+  void Initialize(
+      arma::Mat<eT>& W,
+      const size_t rows,
+      const size_t cols,
+      typename std::enable_if_t<std::is_same<Mode, FanIn>::value>* = 0)
+  {
+    double limit = sqrt(3 * scalingFactor / (double) (rows));
+    RandomInitialization init(-limit, limit);
+    init.Initialize(W, rows, cols);
+  }
+
+  template<typename eT, typename Mode = ModeType>
+  void Initialize(
+      arma::Mat<eT>& W,
+      const size_t rows,
+      const size_t cols,
+      typename std::enable_if_t<std::is_same<Mode, FanOut>::value>* = 0)
+  {
+    double limit = sqrt(3 * scalingFactor / (double) (cols));
+    RandomInitialization init(-limit, limit);
+    init.Initialize(W, rows, cols);
+  }
+
+  template<typename eT, typename Mode = ModeType>
+  void Initialize(
+      arma::Mat<eT>& W,
+      const size_t rows,
+      const size_t cols,
+      typename std::enable_if_t<std::is_same<Mode, FanAvg>::value>* = 0)
+  {
+    double limit = sqrt(6 * scalingFactor /
+        ((double) (rows) + (double) (cols)));
+    RandomInitialization init(-limit, limit);
+    init.Initialize(W, rows, cols);
+  }
+
```

Comment:
Picky style comment, remove the extra line.

Reply:
sure